### ISSUE TITLE
NaN g-2 in the CMSSMCPV

### ISSUE THREAD

**Issue by wkotlarski** [FlexibleSUSY/FlexibleSUSY#485]
This contribution to FFV
```
val += std::complex<double> {1., 0.} * FFV_VSF<Fe,Fe,VP,VWm,Hpm,Fv>::value(indices1, indices2, context, discard_SM_contributions);
```
gives NaN for g-2 (but not edm) in the `CMSSMCPV` default parameter point
```
Block FlexibleSUSYLowEnergy Q= 9.34088556E+02
    21                NAN   # Delta(g-2)/2 of Fe(2) (calculated with FlexibleSUSY)
    23    -3.21031283E-13   # electric dipole moment of Fe(1) [1/GeV]
    24    -6.63800366E-11   # electric dipole moment of Fe(2) [1/GeV]
    25    -1.12092268E-09   # electric dipole moment of Fe(3) [1/GeV]
```

**Comment by wkotlarski** [comment]
It's because `CMSSMCPV` generates a rather large (~2%) imaginary part to $F_2$ form factors.

**Comment by Expander** [comment]
Per definition $a_\mu$ is a real part. So, even if the form factors have a non-zero imaginary part, we should simply take the real part of the result.

**Comment by wkotlarski** [comment]
I'm not sure if it's a real part. I think that it must **only** have a real part, otherwise the higher dimensional operator is not hermitian.

**Comment by wkotlarski** [comment]
This imaginary part is generated `FFV_VSF` and `FFV_SVF` diagrams with `Hpm`, `VWm` and `Fv`.

**Comment by Expander** [comment]
> I'm not sure if it's a real part. I think that it must **only** have a real part, otherwise the higher dimensional operator is not hermitian.

I think we mean the same. :) From looking at the structure, for example in Eqs.(46)-(47) in https://arxiv.org/pdf/hep-ph/0609168.pdf the contribution is purely real. (This is what I meant.) So, I'd argue that if the result is not real, then there is an issue with the generated expression.

My (a bit brisk) proposal was to just take the real part (and remove the check), because the result should be real anyway. But if we really find a non-zero imaginary part, then I still think there is something wrong, sorry.

**Comment by pathron** [comment]
By the way it really won't influence the g-2 paper results.  There were in the end  no models where the vector diagrams were giving the large corrections. I think in the three field cases there were BSM diagrams with the W and the new fermions but these were not the one giving the big (chirality enhanced effect) and we did check carefully that the features in our results could be approximately understood with the simple approximations just using the chirality flipping enhanced contribution.  FS wasn't used for SUSY  (GM2Calc was) so there its also a non issue.

**Comment by wkotlarski** [comment]
Fixed by 5dc8fdd0368d288e279ac7b6ff082a84d7d7b92d.
